compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2319,7 +2327,7 @@ private STNode parseTypeDescriptorInternal(SyntaxKind tokenKind, ParserRuleConte
                     return solution.recoveredNode;
                 }
 
-                return parseTypeDescriptorInternal(solution.tokenKind, context);
+                return parseTypeDescriptorInternal(solution.tokenKind, context, false);
```

Comment:
Shouldn't we pass `isInConditionalExpr` as the last arg?

Reply:
yh It should be. thanks. Fixed with bbc37f6. 
btw just found out that parser crashes for
```
function main() {
    int x = a?b:c
}
```
missing semicolon in conditional-expr. I'll have a look

Reply:
Latter issue fixed with aecf63b